MoviesApp/MoviesApp/ViewController.swift

```diff
@@ -7,13 +7,74 @@
 
 import UIKit
 
-class ViewController: UIViewController {
 
+class ViewController: UIViewController, MovieListView {
+    
+    var myCollectionView: UICollectionView?
+    
+    var presenter: MovieListPresenter?
+    
+    var searchModel: [SearchModel] = []
+    
     override func viewDidLoad() {
         super.viewDidLoad()
-        // Do any additional setup after loading the view.
+        
+        let view = UIView()
+        view.backgroundColor = .gray
+        
+        let mosaicLayout = MosaicLayout()
+        
+        myCollectionView = UICollectionView(frame: self.view.bounds, collectionViewLayout: mosaicLayout)
+        myCollectionView?.backgroundColor = UIColor.clear
+    
+        myCollectionView?.dataSource = self
+        myCollectionView?.delegate = self
+        
+        myCollectionView?.register(MosaicCell.self, forCellWithReuseIdentifier: MosaicCell.identifer)
+        
+        view.addSubview(myCollectionView ?? UICollectionView())
+        self.view = view
+        
+        presenter = MovieListPresenterImpl(view: self, interactor: MovieListInteractorImpl(alomafireProvider: AlomafireProvider()))
+        presenter?.viewDidLoad()
     }
+    
```

Comment:
move protocol methods to extension

extension ViewController: MovieListViewProtocol {
...
}